sw/device/lib/dif/register.h

```diff
@@ -0,0 +1,49 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef SW_DEVICE_LIB_DIF_REGISTER_H_
+#define SW_DEVICE_LIB_DIF_REGISTER_H_
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+typedef volatile uint32_t *reg32_t;
+
+inline reg32_t reg32_new(uintptr_t address) { return (reg32_t)address; }
+
+inline uint32_t reg32_read(reg32_t base, ptrdiff_t offset) {
+  return *(base + offset);
+}
+
+inline void reg32_write(reg32_t base, ptrdiff_t offset, uint32_t value) {
+  *(base + offset) = value;
+}
```

Comment:
I worry that `offset` is misleading, given that it's really `4 * offset` that gets read.  What do you think about renaming to `index` and documenting heavily?

Reply:
Oh. That's a problem. offsets should be in bytes, I think...